src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts

```diff
@@ -175,7 +175,14 @@ export class BatchDetailsComponent implements OnInit, OnDestroy {
           });
           // If batch length is 1, then directly join the batch
           if (this.allBatchList && this.allBatchList.length === 1) {
-            this.enrollBatch(this.allBatchList[0]);
+            const batchStartDate = new Date(this.allBatchList[0]['startDate']);
```

Comment:
why do we need the same chk in this and also int he enroll batch call?

Reply:
If the course has only one batch then we are not opening the Pop up and we are performing direct enrollment to the course for that ongoing batch but if That one batch is future batch then we are checking here.

Reply:
Got It , I update the PR